src/self-managed-debug/src/system_catalog_dumper.rs

```diff
@@ -206,3 +217,577 @@ pub fn spawn_sql_port_forwarding_process(
+    },
+    Relation {
+        name: "mz_scheduling_elapsed",
+        category: RelationCategory::Introspection,
+    },
+    Relation {
+        name: "mz_scheduling_parks_histogram_per_worker",
+        category: RelationCategory::Introspection,
+    },
+    Relation {
+        name: "mz_scheduling_parks_histogram",
+        category: RelationCategory::Introspection,
+    },
+    Relation {
+        name: "mz_compute_lir_mapping_per_worker",
+        category: RelationCategory::Introspection,
+    },
+    Relation {
+        name: "mz_lir_mapping",
+        category: RelationCategory::Introspection,
+    },
+    Relation {
+        name: "mz_compute_error_counts",
+        category: RelationCategory::Introspection,
+    },
+    Relation {
+        name: "mz_compute_error_counts_per_worker",
+        category: RelationCategory::Introspection,
+    },
+    // Relations that are redundant with some of the above, but
+    // are used by the Console.
+    Relation {
+        name: "mz_cluster_replica_metrics_history",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_webhook_sources",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_cluster_replica_history",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_source_statistics",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_cluster_deployment_lineage",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_show_indexes",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_relations",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_frontiers",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_console_cluster_utilization_overview",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_columns",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_kafka_sources",
+        category: RelationCategory::Basic,
+    },
+    Relation {
+        name: "mz_kafka_sinks",
+        category: RelationCategory::Basic,
+    },
+];
+
+const TIMEOUT: Duration = Duration::from_secs(60);
+
+#[derive(Debug, Clone)]
+pub struct ClusterReplica {
+    // TODO (debug_tool1): Use this when scraping introspection relations
+    pub _cluster_name: String,
+    pub _replica_name: String,
+}
+
+pub struct SystemCatalogDumper<'n> {
+    context: &'n Context,
+    pg_client: Arc<PgClient>,
+    // TODO (debug_tool1): Use this when scraping introspection relations
+    _cluster_replicas: Vec<ClusterReplica>,
+    _pg_conn_handle: JoinHandle<Result<(), tokio_postgres::Error>>,
+}
+
+impl<'n> SystemCatalogDumper<'n> {
+    pub async fn new(context: &'n Context, connection_string: &str) -> Result<Self, anyhow::Error> {
+        let (pg_client, pg_conn) = retry::Retry::default()
```

Comment:
FWIW we have our own `PostgresClient` [here](https://github.com/MaterializeInc/materialize/blob/main/src/postgres-client/src/lib.rs#L94-L96) that handles a lot of nice things like connection pooling.

Not necessary to use now but might be worth taking a look!